test/e2e/restart.go

```diff
@@ -0,0 +1,367 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	// How long each node is given during a process that restarts all nodes
+	// before the test is considered failed. (Note that the total time to
+	// restart all nodes will be this number times the nubmer of nodes.)
+	restartPerNodeTimeout = 5 * time.Minute
```

Comment:
Yes, I agree.   I think the best we can do is:

1. Be default, only restart two nodes, in serial (i.e. devs do this by default, as do standard Jenkins runs)
2. In addition, we can try restarting all nodes in parallel, but I'm guessing that this is unlikely to be successful, and it's fairly unrepresentative of reality anyway.  Make it possible to do this via appropriate environment variables, and we can dedicate a Jenkins job to this.
3. Make it possible to run this test completely (i.e. restart every node, and in serial) using environment variables, and dedicate a Jenkins test run to this. This is the most realistic test, but unfortunately it takes too long to have it delay the standard e2e test runs.